### ISSUE TITLE
[1.7.0] After setting up kubeadm, the master node has status NotReady

### ISSUE THREAD

**Issue by piersbarrios** [kubernetes/kubernetes#48371]
/kind bug

**What happened**:

After setting up kubeadm, the master node has status NotReady

**What you expected to happen**:

master node has a ready status.

**How to reproduce it (as minimally and precisely as possible)**:

```
vagrant up && vagrant ssh master
sudo su
yum install docker -y && systemctl enable docker && systemctl start docker 
```

(from the official kubeadm tutorial)
```
cat <<EOF > /etc/yum.repos.d/kubernetes.repo
[kubernetes]
name=Kubernetes
baseurl=https://packages.cloud.google.com/yum/repos/kubernetes-el7-x86_64
enabled=1
gpgcheck=1
repo_gpgcheck=1
gpgkey=https://packages.cloud.google.com/yum/doc/yum-key.gpg
        https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg
EOF
setenforce 0
yum install -y kubelet kubeadm
systemctl enable kubelet && systemctl start kubelet
```

```
kubeadm init --apiserver-advertise-address 10.10.10.10
mkdir -p $HOME/.kube
sudo cp -i /etc/kubernetes/admin.conf $HOME/.kube/config
sudo chown $(id -u):$(id -g) $HOME/.kube/config
```

`kubectl get nodes`

NAME      STATUS     AGE       VERSION
k8t-m     NotReady   7m        v1.7.0

**Anything else we need to know?**:

My Vagrantfile:
```
# -*- mode: ruby -*-
# vi: set ft=ruby :
VAGRANTFILE_API_VERSION = "2"
Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
 config.vm.define :master, primary: true do |master_config|
  master_config.vm.box = "maier/centos-7.3.1611-x86_64"  
  master_config.vm.hostname = "K8t-M"
  master_config.vm.network "private_network", ip: "10.10.10.10"
 end
 config.vm.define :minion1 do |minion_config|
  minion_config.vm.box = "maier/centos-7.3.1611-x86_64"
  minion_config.vm.hostname = "K8t-Node"
  minion_config.vm.network "private_network", ip: "10.10.10.11"
 end
end
```

**Environment**:
- Kubernetes version (use `kubectl version`): 1.7.0
- Cloud provider or hardware configuration**: using vagrant on a MacBook Pro
- OS (e.g. from /etc/os-release): Centos 7 
- Kernel (e.g. `uname -a`): Linux K8t-M 3.10.0-514.2.2.el7.x86_64 #1 SMP Tue Dec 6 23:06:41 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: See instructions above
- Others:

**Comment by piersbarrios** [comment]
@kubernetes/sig-docs

**Comment by piersbarrios** [comment]
Extra information: Since the kubernetes documentation got updated this weekend , I retried a full install and same problem. 

With kubelet logs I get: 

`error: failed to run Kubelet: failed to create kubelet: misconfiguration: kubelet cgroup driver: "cgroupfs" is different from docker cgroup driver: "systemd`

```
# cat /etc/systemd/system/kubelet.service.d/10-kubeadm.conf 
[Service]
Environment="KUBELET_KUBECONFIG_ARGS=--kubeconfig=/etc/kubernetes/kubelet.conf --require-kubeconfig=true"
Environment="KUBELET_SYSTEM_PODS_ARGS=--pod-manifest-path=/etc/kubernetes/manifests --allow-privileged=true"
Environment="KUBELET_NETWORK_ARGS=--network-plugin=cni --cni-conf-dir=/etc/cni/net.d --cni-bin-dir=/opt/cni/bin"
Environment="KUBELET_DNS_ARGS=--cluster-dns=10.96.0.10 --cluster-domain=cluster.local"
Environment="KUBELET_AUTHZ_ARGS=--authorization-mode=Webhook --client-ca-file=/etc/kubernetes/pki/ca.crt"
Environment="KUBELET_CADVISOR_ARGS=--cadvisor-port=0"
Environment="KUBELET_CGROUP_ARGS=--cgroup-driver=systemd"
ExecStart=
ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_SYSTEM_PODS_ARGS $KUBELET_NETWORK_ARGS $KUBELET_DNS_ARGS $KUBELET_AUTHZ_ARGS $KUBELET_CADVISOR_ARGS $KUBELET_CGROUP_ARGS $KUBELET_EXTRA_ARGS
```

I think what what I'm seeing here is what ReSearchITEng describes in https://github.com/kubernetes/kubernetes/issues/43819

**Comment by danodob** [comment]
Actually what ReSearchITEng is referring to in [#43819](https://github.com/kubernetes/kubernetes/issues/43819)  is the reverse cgroup driver configuration. It is the same thing I am running into as well. Docker is using cgroupfs whilst kubelet is using systemd. Every other issue/thread I have read references the opposite.

**Comment by piersbarrios** [comment]
This can be closed as going onto the 1.7.1 (with other issues though)

**Comment by shreeshailgumgeri** [comment]
I too had same problem.
Below command solved my issues. Just try out this
systemctl enable kubelet && systemctl start kubelet
